tests/ferret/core_test.go

```diff
@@ -298,6 +342,93 @@ func TestCore(t *testing.T) {
 						`consider using $getField or $setField.`,
 				},
 			},
+
+			// binary
+
+			// $eq
+			{
+				name: "EqBinary",
+				q:    bson.D{{"value", bson.D{{"$eq", primitive.Binary{Subtype: 0x80, Data: []byte{42, 0, 13}}}}}},
+				IDs:  []string{"binary"},
+			},
+
+			// bool
+
+			// $eq
+
+			{
+				name: "EqBoolTrue",
+				q:    bson.D{{"value", bson.D{{"$eq", true}}}},
+				IDs:  []string{"bool-true"},
+			},
+			{
+				name: "EqBoolFalse",
+				q:    bson.D{{"value", bson.D{{"$eq", false}}}},
+				IDs:  []string{"bool-false"},
+			},
+
+			// datetime
+
+			// $eq
+
+			{
+				name: "EqDatetime",
+				q:    bson.D{{"value", bson.D{{"$eq", time.Date(2021, 11, 1, 10, 18, 42, 123000000, time.UTC)}}}},
+				IDs:  []string{"datetime"},
+			},
+
+			// timestamp
+
+			// $eq
+
+			{
+				name: "EqTimestamp",
+				q:    bson.D{{"value", bson.D{{"$eq", primitive.Timestamp{T: 42, I: 13}}}}},
+				IDs:  []string{"timestamp"},
+			},
+
+			// null
+
+			{
+				name: "EqNull",
+				q:    bson.D{{"value", bson.D{{"$eq", nil}}}},
+				IDs:  []string{"null"},
+			},
+
+			// regex
+
+			// $eq
+
+			{
+				name: "EqRegex",
+				q:    bson.D{{"value", primitive.Regex{Pattern: "foo", Options: "i"}}},
```

Comment:
That test is important, but it does not actually test `$eq` operator. We should add another one that uses it as it is the place where regex is somewhat special.